src/elements/Chip/Chip.tsx

```diff
@@ -25,6 +26,11 @@ export interface ChipProps extends React.HTMLAttributes<HTMLSpanElement> {
    */
   variant?: 'filled' | 'outline';
 
+  /**
+   * If there is an icon for the chip.
+   */
+  icon?: React.ReactElement;
```

Comment:
I'm kind of on the fence with this. 

On one hand, it makes it much easier for the user to define something like `icon` and it handle it for you. On the other hand, it would be more flexible to have it like `start` and `end` like I did with `Input` to make it more generic.

I think a nice compromise might be making a `DeleteableChip` or something like that and encompassing this logic there. Open to thoughts though.

Reply:
hmm `start` instead of `icon` makes sense. but not sure why we want an entirely different component for deletable though.. Maybe just omit `end` in this case and show the delete button if `onDelete` is defined?

Reply:
So, my thought is... delete sounds like a very specific use case. Chips are pretty generic and don't nesc always need a delete. By adding a delete interface it bloats the api of the core component for a use case that might not be that common. On the other hand, by making it more generic and then creating a specific 'wrapper' around it, it makes it more obvious and more flexible. I saw that MUI does this and has one for avatar and such as well. But again those are implementation details of the chip, rather than chip functions.

Reply:
ok - went ahead and separated it .. let me know what you think